src/test/java/com/michelin/cio/hudson/plugins/maskpasswords/MaskPasswordsWorkflowTest.java

```diff
@@ -99,19 +101,27 @@ void basics(JenkinsRule j) throws Throwable {
         p.setDefinition(new CpsFlowDefinition("node {wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [[var: 'PASSWORD', password: 's3cr3t']]]) {semaphore 'waiting'; echo 'printed s3cr3t oops'}}", true));
         WorkflowRun b = p.scheduleBuild2(0).waitForStart();
         SemaphoreStep.waitForStart("waiting/1", b);
-        Set<String> expected = new HashSet<>(Arrays.asList("build.xml", "program.dat", "workflow/5.xml"));
+        Set<String> expected1 = new HashSet<>(Arrays.asList("build.xml", "program.dat", "workflow/5.xml"));
         if (!Functions.isWindows()) {
             // Skip assertion on Windows, temporary files contaminate content frequently
-            assertEquals(expected, grep(b.getRootDir(), "s3cr3t"), "TODO cannot keep it out of the closure block, but at least outside users cannot see this; withCredentials does better");
+            await().atMost(5, TimeUnit.SECONDS)
+                    .alias("TODO cannot keep it out of the closure block, but at least outside users cannot see this; withCredentials does better")
+                    .until(() ->
+                            expected1.equals(grep(b.getRootDir(), "s3cr3t"))
+                    );
```

Comment:
This does not print the actual value if the test fails, like it did before this PR. Hamcrest matchers can restore this:

```java
.until(() -> grep(b.getRootDir(), "s3cr3t"), equalTo(expected));
```